src/main/java/soot/Scene.java

```diff
@@ -1217,16 +1217,24 @@ public SootClass getSootClassUnsafe(String className) {
   public SootClass getSootClassUnsafe(String className, boolean phantomNonExist) {
     RefType type = nameToClass.get(className);
     if (type != null) {
-      SootClass tsc = type.getSootClass();
-      if (tsc != null) {
-        return tsc;
+      synchronized (type) {
+        if (type.hasSootClass()
+            || !SootClass.INVOKEDYNAMIC_DUMMY_CLASS_NAME.equals(className)) {
+          SootClass tsc = type.getSootClass();
```

Comment:
Why doesn't it render a problem if a type does not have a soot class as long as it is the `INVOKEDYNAMIC_DUMMY_CLASS_NAME `. 

Can't we find a more general way to prevent such issues? The current fix seems rather tailored towards the fact that `AsmMethodSource` creates this dummy class during `retrieveAllBodies`. Or do I misinterpret this fix?